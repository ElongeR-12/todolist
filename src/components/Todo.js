import React from 'react'
import { Link } from 'react-router-dom'
const Todo = ({todo, setTodos, todos}) => {
    const toggleCheckbox = (id) =>{
        const newTodos = [...todos];
        const todo = newTodos.filter((item)=>item.id===id)
        todo[0].state = !todo.state;
        const remainTodo = newTodos.filter((item)=>item.id !== id);
        const todoFinal = [...remainTodo, ...todo]
        setTodos([...todoFinal]);
      }
  return (
    <div>
        <p>
          <input
            type="checkbox"
            defaultChecked={todo?.state}
            onClick={()=>toggleCheckbox(todo.id)}
          />
        <span>{todo?.title}</span>
        <Link to={`/todolist/todo/${todo.id}`}>details</Link>
        </p>
    </div>
  )
}

export default Todo