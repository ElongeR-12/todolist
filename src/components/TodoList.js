import React from 'react'
import Todo from './Todo'

function TodoList({todos, setTodos}) {
    
  return(
      <>
        <h1>Todo List</h1>
        { 
            todos && todos.map((todo)=>(<Todo key={todo.id} todo={todo} setTodos={setTodos} todos={todos}/>))
        }
      </>
  )
  
}

export default TodoList