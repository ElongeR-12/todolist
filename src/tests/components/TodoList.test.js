import React from 'react';
import { shallow } from 'enzyme';
import Todo from '../../components/Todo';
import TodoList from '../../components/TodoList';
describe('list todo',()=>{
    test('should contain Todo component',()=>{
        const wrapper = shallow(<TodoList/>);
        expect(wrapper.find(Todo)).toBeTruthy()
    })
    test('shoud return one todo',()=>{
        const todos = [{
            id: 1,
            state:false,
            description:"Lorem ipsum",
            title:"title three"
        }]
        const wrapper = shallow(<Todo todos={todos}/>);
        expect(wrapper).toMatchSnapshot();
    })
});