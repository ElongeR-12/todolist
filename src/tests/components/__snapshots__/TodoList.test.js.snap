// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list todo shoud return on todo 1`] = `
<div>
  <p>
    <input
      defaultChecked={false}
      onClick={[Function]}
      type="checkbox"
    />
    <span>
      title three
    </span>
  </p>
</div>
`;
